udp/udp_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2015 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package udp
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/chihaya/chihaya/config"
+	"github.com/chihaya/chihaya/stats"
+	"github.com/chihaya/chihaya/tracker"
+
+	_ "github.com/chihaya/chihaya/backend/noop"
+)
+
+var testPort = "34137"
+var connectAction = []byte{0, 0, 0, byte(connectActionID)}
+var announceAction = []byte{0, 0, 0, byte(announceActionID)}
+var scrapeAction = []byte{0, 0, 0, byte(scrapeActionID)}
+var errorAction = []byte{0, 0, 0, byte(errorActionID)}
+
+func init() {
+	stats.DefaultStats = stats.New(config.StatsConfig{})
```

Comment:
I'd be okay for either, but I feel like the interface is the better solution as is enforced by the compiler.